ts-sdk/whirlpool/src/swap.ts

```diff
@@ -0,0 +1,227 @@
+    getTickArrayAddress(whirlpool.address, tickArrayStartIndex).then(
+      (x) => x[0],
+    ),
+    getTickArrayAddress(whirlpool.address, tickArrayStartIndex + offset).then(
+      (x) => x[0],
+    ),
+    getTickArrayAddress(
+      whirlpool.address,
+      tickArrayStartIndex + offset * 2,
+    ).then((x) => x[0]),
+    getTickArrayAddress(whirlpool.address, tickArrayStartIndex - offset).then(
+      (x) => x[0],
+    ),
+    getTickArrayAddress(
+      whirlpool.address,
+      tickArrayStartIndex - offset * 2,
+    ).then((x) => x[0]),
+  ]);
+
+  const tickArrays = await fetchAllTickArray(
+    rpc,
+    tickArrayAddresses,
+  );
+  const oracleAddress = await getOracleAddress(whirlpool.address).then(
+    (x) => x[0],
+  );
+
+  const currentEpoch = await rpc.getEpochInfo().send();
+  const transferFeeA = getCurrentTransferFee(tokenA.data, currentEpoch.epoch);
+  const transferFeeB = getCurrentTransferFee(tokenB.data, currentEpoch.epoch);
+
+  const quote = getSwapQuote<T>(
+    params,
+    whirlpool.data,
+    transferFeeA,
+    transferFeeB,
+    tickArrays,
+    specifiedTokenA,
+    slippageTolerance,
+  );
+  const maxInAmount = "tokenIn" in quote ? quote.tokenIn : quote.tokenMaxIn;
+  const tokenAIsInput = specifiedTokenA === specifiedInput;
+
+  const { createInstructions, cleanupInstructions, tokenAccountAddresses } =
+    await prepareTokenAccountsInstructions(rpc, signer, {
+      [whirlpool.data.tokenMintA]: tokenAIsInput ? maxInAmount : 0n,
+      [whirlpool.data.tokenMintB]: tokenAIsInput ? 0n : maxInAmount,
+    });
+
+  const instructions: IInstruction[] = [];
+
+  instructions.push(...createInstructions);
+
+  const specifiedAmount =
+    "inputAmount" in params ? params.inputAmount : params.outputAmount;
+  const otherAmountThreshold =
+    "tokenMaxIn" in quote ? quote.tokenMaxIn : quote.tokenMinOut;
+
+  const swapInstruction = getSwapV2Instruction({
+    tokenProgramA: tokenA.programAddress,
+    tokenProgramB: tokenB.programAddress,
+    memoProgram: MEMO_PROGRAM_ADDRESS,
+    tokenAuthority: signer,
+    whirlpool: whirlpool.address,
+    tokenMintA: whirlpool.data.tokenMintA,
+    tokenMintB: whirlpool.data.tokenMintB,
+    tokenOwnerAccountA: tokenAccountAddresses[whirlpool.data.tokenMintA],
+    tokenOwnerAccountB: tokenAccountAddresses[whirlpool.data.tokenMintB],
+    tokenVaultA: whirlpool.data.tokenVaultA,
+    tokenVaultB: whirlpool.data.tokenVaultB,
+    tickArray0: tickArrays[0].address,
+    tickArray1: tickArrays[1].address,
+    tickArray2: tickArrays[2].address,
+    amount: specifiedAmount,
+    otherAmountThreshold,
+    sqrtPriceLimit: 0,
+    amountSpecifiedIsInput: specifiedInput,
+    aToB: specifiedTokenA,
+    oracle: oracleAddress,
+    remainingAccountsInfo: {
+      slices: [
+        { accountsType: AccountsType.SupplementalTickArraysOne, length: 32 },
+        { accountsType: AccountsType.SupplementalTickArraysTwo, length: 32 },
+      ],
```

Comment:
for `swapV2` valid type is `AccountsType.SupplementalTickArray` (One and Two are for twoHopSwapV2).

We provide 2 supplemental tickarrays, so `length` is 2.